pain-points.md

```diff
@@ -0,0 +1,21 @@
+# Ouch
+
+## Stuff that are actually wrong
+
+- `store/home` is also defined in `vtex.store`, we should tell people they can extend that, but we maybe should also incentivize them to extend `vtex.dreamstore` whenever possible. What now?
+- It's hard to explain that we can extend a block `a` by declaring `a` in another file without the need to call it `a/b`.
+- We don't have `vtex.header` and `vtex.footer`, we only have `vtex.dreamstore-header` and `vtex.dreamstore-footer`. Should we declare `header` and footer within `vtex.store`?
+
+## Blocks
+
+- `extensions` -> `blocks`
+- `templates` -> `blocks`
+- `ExtensionPoint` -> `block`
+
+## Dependencies
+
+- They will forget to depend on what they want to extend, creating useless blocks.
```

Comment:
Usually, I think the need for extension is only on `store` modules (like `shelf`, `header`, `banner`, etc). Can you think of anything else, aside from tailor made react components?

Reply:
As I've responded in a comment above, we're thinking of using them in `store`, not defining them in `store`. The idea was to spread the definitions, even for basic blocks, across different apps, `vtex.shelf`, `vtex.header`, etc.

This bit is specially polemic, I'd appreciate suggestions.